src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs

```diff
@@ -281,7 +281,11 @@ private void UpdateSyncModes(SyncMode newModes, string? reason = null)
             Preparing?.Invoke(this, args);
             Changing?.Invoke(this, args);
             Current = newModes;
-            Changed?.Invoke(this, args);
+            if (Changed is not null)
+            {
+                Parallel.ForEach(Changed.GetInvocationList(),
+                    deleg => deleg.DynamicInvoke(this, args));
+            }
```

Comment:
I think it should not happen, but in theory pattern is to do a local copy to avoid `NullReferenceException` and be thread-safe:

https://stackoverflow.com/questions/33166143/local-copy-of-event

Reply:
😲 You learn something every day